src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala

```diff
@@ -468,6 +555,34 @@ class CassandraReadJournal(system: ExtendedActorSystem, config: Config)
       EventEnvelope(offset, persistentRepr.persistenceId, persistentRepr.sequenceNr, payload)
     }
 
+  private[this] def toEventEnvelope2(persistentRepr: PersistentRepr, offset: Offset): immutable.Iterable[EventEnvelope2] =
+    adaptFromJournal(persistentRepr).map { payload =>
+      EventEnvelope2(offset, persistentRepr.persistenceId, persistentRepr.sequenceNr, payload)
+    }
+
+  private[this] def eventEnvelope2toUuidEnvelope(env: EventEnvelope2): UUIDEventEnvelope =
+    UUIDEventEnvelope(
+      env.offset match {
+        case TimeBasedUUID(uuid) => uuid
+        case unexpected          => throw new IllegalArgumentException(s"Unexpected offset type $unexpected for envelope $env")
+      },
+      env.persistenceId,
+      env.sequenceNr,
+      env.event
+    )
+
+  private[this] def internalUuidToOffset(uuid: UUID): Offset =
+    if (uuid == firstOffset) NoOffset
+    else TimeBasedUUID(uuid)
+
+  private[this] def offsetToInternalOffset(offset: Offset): UUID =
+    offset match {
+      case TimeBasedUUID(uuid) => uuid
+      case NoOffset            => firstOffset
+      case unsupported =>
```

Comment:
Just to make sure I'm clear on the semantics here, the user makes a query with `Sequence(value)` (where value is a timestamp and not a sequence number), the returned envelopes then have the `TimeBasedUUID` for each event or do we do a lossy (loosing the rest of the uuid except for the timestamp) transformation back based on the original query type?

Also, isn't this terribly confusing compared to only accepting `TimeBasedUUID` and the user can create the uuid using `UUID.fromTime(timestamp)` themselves?